### PR TITLE
Improve Script efficiency by Cleaning source code

### PR DESCRIPTION
**Pull request by zakariaGatter**
Just some Syntax improvment and fixes plus removing some unnecessary commands like `awk {print $1}` and others

### CODE REVIEW

File: src/layout.sh [hunk 1]
```diff
@@ -158,27 +157,27 @@ start_listener() {
   __recalculate_layout() { run_layout $layout $args 2> /dev/null || true; }
 
   # Then listen to node changes and recalculate as required
-  bspc subscribe node_{add,remove,transfer,flag,state} desktop_focus | while read line; do
-    event=$(echo "$line" | awk '{print $1}')
-    arg_index=$([[ "$event" == "node_transfer" ]] && echo "6" || echo "3")
-    desktop_id=$(echo "$line" | awk "{print \$$arg_index}")
+  bspc subscribe node_{add,remove,transfer,flag,state} desktop_focus | while read -a line; do
```
Comment (phenax) [inline src/layout.sh:160]: Didn't know you could read it as an array of words. This is pretty neat!

**Review by phenax — COMMENTED** [review]
This looks pretty good!

File: src/layout.sh (hunk 1 above)
Comment (zakariaGatter) [reply src/layout.sh:160]: with bash `read` you can do much more

File: src/layout.sh [hunk 2]
```diff
@@ -28,15 +28,18 @@ remove_listener() {
   kill_layout "$desktop"
 
   # Reset process id and layout
-  set_desktop_option $desktop 'layout' ""
-  set_desktop_option $desktop 'pid'    ""
+  set_desktop_option $desktop 'layout'
```
[the remark this answers is not in the document]
Comment (amtoine) [reply src/layout.sh:31]: i agree with that comment!

even if the argument has no value, `set_desktop_option` can take one here, that looks clearer to me :yum:

File: src/layout.sh [hunk 3]
```diff
@@ -28,15 +28,18 @@ remove_listener() {
...
+  set_desktop_option $desktop 'pid'
 }
 
 # layout -> filename
 get_layout_file() {
   local layout_file="$LAYOUTS/$1.sh"
   # GUARD: Check if layout exists
-  [[ ! -f $layout_file ]] && echo "Layout [$layout_file] does not exist" && exit 1
+  if [ -f $layout_file ]; then
```
Comment (amtoine) [inline src/layout.sh:39]: actually, that is wrong :thinking: 

i tried
```bash
if [ -f $file ]; then echo "does not exist"; fi
```
and
```bash
[[ ! -f $file ]] && echo "does not exist"
```
and i do not get the same output...

the condition is simply inverted

please apply

```diff
-  if [ -f $layout_file ]; then
+  if [ ! -f $layout_file ]; then
```
:yum:

Comment (amtoine) [reply src/layout.sh:39]: and that is actually a major issue! :open_mouth: 

in the current state of the PR, `bsp-layout` does not work anymore :confused: 

however, applying the patch above fixes the issue and the layouts work again :wink:

**Review by amtoine — CHANGES_REQUESTED** [review]
Nice work on the PR :yum: 

However, there is a major issue that needs to be addressed as `bsp-layout` does not work for now on the tip of the PR branch.

The change is small and the PR will be ready soon :relieved:

File: src/layout.sh (hunk 3 above)
Comment (phenax) [reply src/layout.sh:39]: Nice catch!

Comment (zakariaGatter) [reply src/layout.sh:39]: really a nice catch this totaly gone over my here 

sorry

Comment (amtoine) [reply src/layout.sh]: thanks for the bug fix :wink:

File: src/layout.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (amtoine) [reply src/layout.sh:31]: only that change and i think we'll be good to merge @zakariaGatter :wink:

[the remark this answers is not in the document]
Comment (zakariaGatter) [reply src/layout.sh:31]: ok ill work on it, thank you

[the remark this answers is not in the document]
Comment (amtoine) [reply src/layout.sh]: this looks great to me :ok_hand:

**Review by amtoine — APPROVED** [review]

**Comment by amtoine** [comment]
That looks fine to me :yum: :+1:

**Pull request merged**